### PR TITLE
Fix crash when JSON response contains pagination key but with a null value

### LINKED ISSUE #663: Crash when pagination value is null in JSON response
**Describe the bug**

The following program crashed, since sometime the server returns JSON data which are not correctly handle by the `SearchIterator` class in `explore_iterator.py`.

**To Reproduce**

I'm calling the API endpoint (`/api/v3/findings/vulnerabilities/webapp/search`) on cloud.tenable.com, using `tio.v3.explore.findings.search_webapp` function with some arguments:
```json
{
    "limit": 200,
    "filter": {
        "and": [
            {
                "operator": "eq",
                "value": "redacted",
                "property": "asset_id"
            }
        ]
    },
    "fields": [
        "asset",
        "definition.id",
        "definition.vpr.score",
        "definition.exploitability_ease"
    ],
    "next": "**redacted**"
}
```
The server responds (sometimes):
```json
{"findings":[**redacted**],"pagination":null}
```

Since the python code is:
```python
def _process_response(self, response: Response) -> None:
    '''  
    Process the API Response
    '''
    body = response.json()
    pagination = body.get('pagination', {})
    self.page = body[self._resource]
    self.total = pagination.get('total')
    self._next_token = pagination.get('next')
```
The key `pagination` actually exists in JSON response, so `pagination` variable will be `None` and not a dict.
Crash happens line 8 (line 72 in the actual code). 

```
Traceback (most recent call last):
  ***redacted***
  File "/redacted/lib/python3.10/site-packages/restfly/iterator.py", line 114, in __next__
    return self.next()  # noqa: PLE1102
  File "/redacted/lib/python3.10/site-packages/restfly/iterator.py", line 140, in next
    self._get_page()
  File "/redacted/lib/python3.10/site-packages/tenable/io/v3/base/iterators/explore_iterator.py", line 59, in _get_page
    self._process_response(resp)
  File "/redacted/lib/python3.10/site-packages/tenable/io/v3/base/iterators/explore_iterator.py", line 74, in _process_response
    self.total = pagination.get('total')
AttributeError: 'NoneType' object has no attribute 'get'
```

**Expected behavior**
The code should not crashed.

**System Information (please complete the following information):**
 - OS: Linux
 - Architecture 64bit

### PR DESCRIPTION
**Pull request by kalidor**
# Description

Quick fix to avoid crashing when pagination key is in HTTP response but with a null value.

Fixes #663

## Type of change

Please delete options that are not relevant.

- [X ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Just re-run the program using a proxy to see the HTTP response containing `{"findings":[**redacted**],"pagination":null}`, and the program does not crash anymore.

# Checklist:

- [ X] My code follows the style guidelines of this project
- [ X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by inayathulla** [comment]
@kalidor thanks for finding the issue along with the fix, please add a test case to validate functionality then we would be good to merge.

**Comment by kalidor** [comment]
This test is enough ? or do i need to create a more complete test like (test_explore_v3.py) ? I'm hesitating...

File: tests/io/test_search_iterator_v3.py [hunk 1]
```diff
@@ -61,3 +65,19 @@ def test_search_iterator_v3(api):
...
+def test_search_iterator_v3_null_pagination(api):
+    '''
+    Test for null pagination in SearchIterator._process_response
+    '''
+    search_iterator = SearchIterator(
+        api=api
+    )
+    class TempJson:
+        def json(self):
+            return json.loads(json.dumps(NULL_PAGINATION_RESPONSE))
```
Comment (SteveMcGrath) [inline tests/io/test_search_iterator_v3.py:79]: I dont think we need to test with a full example here:

```suggestion
            return json.loads(json.dumps({'findings': [{'id': 'abcdef'}],
                                          'pagination': None
                                          })
                              )
```

**Review by SteveMcGrath — APPROVED** [review]

**Comment by kalidor** [comment]
Hi, do you need me to do something else before merging?
Thanks!

**Comment by SteveMcGrath** [comment]
@varap11

**Comment by Rogdham** [comment]
Hello @SteveMcGrath @aseemsavio thank you for the review and the merge! Do you have a timeline about the release of the new version of the library including this fix? Many thanks!

**Comment by aseemsavio** [comment]
We are awaiting some Tenable.ot related changes. This will go out along with it. There isn't a firm timeline for the OT changes yet.

**Comment by Rogdham** [comment]
Thank you! :heart:

**Pull request merged**